src/main/java/com/nike/cerberus/command/core/GenerateCertsCommand.java

```diff
@@ -0,0 +1,177 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nike.cerberus.command.core;
+
+import com.beust.jcommander.Parameter;
+import com.beust.jcommander.Parameters;
+import com.nike.cerberus.command.Command;
+import com.nike.cerberus.operation.Operation;
+import com.nike.cerberus.operation.core.GenerateCertsOperation;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.nike.cerberus.command.core.GenerateCertsCommand.COMMAND_DESCRIPTION;
+import static com.nike.cerberus.command.core.GenerateCertsCommand.COMMAND_NAME;
+
+@Parameters(
+        commandNames = COMMAND_NAME,
+        commandDescription = COMMAND_DESCRIPTION
+)
+public class GenerateCertsCommand implements Command {
+
+    public static final String COMMAND_NAME = "generate-certs";
+    public static final String COMMAND_DESCRIPTION = "Generates the TLS certificates needed to enable https " +
+            "through out the system, using an ACME provider such as LetsEncrypt";
+
+    public static final String BASE_DOMAIN_LONG_ARG = "--base-domain";
+    public static final String BASE_DOMAIN_SHORT_ARG = "-d";
+
+    public static final String HOSTED_ZONE_ID_LONG_ARG = "--hosted-zone-id";
+    public static final String HOSTED_ZONE_ID_SHORT_ARG = "-h";
+
+    public static final String ADDITIONAL_SUBJECT_ALT_NAME_LONG_ARG = "--additional-subject-alternative-name";
+    public static final String ADDITIONAL_SUBJECT_ALT_NAME_SHORT_ARG = "-a";
+
+    public static final String ENABLE_LE_CERTFIX_LONG_ARG = "--enable-letsecrypt-certfix";
+    public static final String ENABLE_LE_CERTFIX_SHORT_ARG = "-elec";
+
+    public static final String CERT_FOLDER_LONG_ARG = "--cert-dir";
+    public static final String CERT_FOLDER_SHORT_ARG = "-c";
+
+    public static final String ACME_API_LONG_ARG = "--acme-api-url";
+    public static final String ACME_API_SHORT_ARG = "-u";
+
+    public static final String LETS_ENCRYPT_ACME_API_URI = "acme://letsencrypt.org";
+
+    public static final String CONTACT_EMAIL_LONG_ARG = "--contact-email";
+    public static final String CONTACT_EMAIL_SHORT_ARG = "-c";
+
+    @Parameter(
+            names = {
+                    BASE_DOMAIN_LONG_ARG,
+                    BASE_DOMAIN_SHORT_ARG
+            },
+            description = "The base domain for the environment that this command will use to generate the following " +
+                    "subject name {env}.{base-domain} and with the following subject alternative name {env}.{region}.{base-domain}\n" +
+                    "ex: cerberus -e demo -r us-west-2 generate-certs -d cerberus-oss.io would make a cert for demo.cerberus-oss.io " +
+                    "with a sans of demo.us-west-2.cerberus-oss.io so that we can create a CNAME record for " +
+                    "demo.cerberus-oss.io that will point to an ALB in us-west-2 with a CNAME record of " +
```

Comment:
I've started writing example domains like: `demo.cerberus.example.com`

I think it is a little better.  example.com is an official domain name for examples.

Reply:
There are multiple places where cerberus-oss.io is used. How about after this gets merged I do a find and replace PR to `cerberus-oss.io` -> `cerberus.example.com`.

Reply:
Sure, anytime.